### DISCUSSION TITLE
Does CLI have a flag for specifying Data Connect Emulator PGLite Database strings ?

### DISCUSSION THREAD

**Discussion by HassanBahati** [firebase/firebase-tools#7997] [General]
My goal is to start the Data Connect Emulator in non-interactive environments such as a **GitHub Action** this is what I found in the documentations [Start the emulator](https://firebase.google.com/docs/data-connect/data-connect-emulator-suite?hl=en&authuser=1#javascript)

I have tried starting a Data Connect Emulator locally by running the command below;
```sh
firebase emulators:start --only dataconnect --project test
```
but fails with the following logs
```
% firebase emulators:start --only dataconnect --project test
i  emulators: Starting emulators: dataconnect
i  dataconnect: downloading dataconnect-emulator-1.3.6...
Progress: ===================================================================================================================================> (100% of 25MB)
i  dataconnect: Removing outdated emulator files: dataconnect-emulator-1.7.0
i  dataconnect: Data Connect Emulator logging to dataconnect-debug.log
i  emulators: Shutting down emulators.
i  dataconnect: Stopping Data Connect Emulator
⚠  Data Connect Emulator has exited upon receiving signal: SIGINT
i  hub: Stopping emulator hub

Error: No Postgres connection string found in '.firebaserc'. The Data Connect emulator will not be able to execute operations.
Run firebase setup:emulators:dataconnect to set up a Postgres connection.
```

Is it that it is a must to have a `.firebaserc` in my directory ? 
seems the `.firebaserc` is meant to contain the Postgres connection string, is there a way to define this using a flag so that i could execute a command in a github action to start the emulator ??

The Documentation at [Use the Data Connect emulator for CI/CD](https://firebase.google.com/docs/data-connect/data-connect-emulator-suite) doesnt seem to talk about a database setup or specification..

I am really curious to know how the Data Connect emulator is meant to be setup to run in non-interactive environments like Github Actions ?

FYI: The Data Connect Emulator starts, runs and works fine when i use the [Firebase Data Connect Emulator VsCode Extension](https://marketplace.visualstudio.com/items?itemName=GoogleCloudTools.firebase-dataconnect-vscode) without a Postgres connection setup in the `.firebaserc` file

**Comment by HassanBahati** [comment]
CC: @joehan

**Comment by joehan** [comment]
Sorry for the slow reply here! You can also provide this value via  the`FIREBASE_DATACONNECT_POSTGRESQL_STRING` env var, like so:

```
export FIREBASE_DATACONNECT_POSTGRESQL_STRING=postgres://localhost:5432/mydatabase?sslmode=disable
firebase emulators:start
```

On a separate note, you'll generally want to use `emulators:exec` when running in noninteractive environments

**Reply by HassanBahati** [reply]
thank you @joehan , i will be sure to try this
